extension/test/unit/config/init/ensure-resources.spec.js

```diff
@@ -0,0 +1,64 @@
+const { expect } = require('chai')
+const sinon = require('sinon')
+const _ = require('lodash')
+const { ensureResources } = require('../../../../src/config/init/ensure-resources')
+
+const webComponentsPaymentType = require('../../../../resources/web-components-payment-type.json')
+const apiExtension = require('../../../../resources/api-extension.json')
+const interfaceInteractionType = require('../../../../resources/payment-interface-interaction-type.json')
+
+describe('Ensure resources', () => {
+  const mockClient = {
+    get builder () {
+      return {
+        extensions: {
+          where: () => {
+          }
+        },
+        types: {
+          where: () => {
+          }
+        }
+      }
+    },
+    fetch () {
+    },
+    create () {
+    }
+  }
+
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('should ensure payment type, interface interaction type and API extension are created', async () => {
+    sinon.stub(mockClient, 'fetch').returns({ body: { results: [] } })
+    const createStub = sinon.stub(mockClient, 'create').returns({ body: { results: [] } })
+
+    await ensureResources(mockClient)
+
+    expect(createStub.callCount).to.equal(3)
+
+    const callArgs = _.flattenDeep(createStub.args)
+    const createdPaymentType = callArgs.find(arg => arg.key === webComponentsPaymentType.key)
+    expect(createdPaymentType).to.deep.equal(webComponentsPaymentType)
+
+    const createdApiExtension = callArgs.find(arg => arg.key === apiExtension.key)
+    // set the url variable in the template to the correctly evaluated value
+    const apiExtensionCloned = _.cloneDeep(apiExtension)
+    apiExtensionCloned.destination.url = ''
+    expect(createdApiExtension).to.deep.equal(apiExtensionCloned)
+
+    const createdInterfaceInteractionType = callArgs.find(arg => arg.key === interfaceInteractionType.key)
+    expect(createdInterfaceInteractionType).to.deep.equal(interfaceInteractionType)
+  })
+
+  it('should skip creating resources when they already exist', async () => {
```

Comment:
you said you removed that catches, and the test should fail, right?

Reply:
No, if the resources already exist, it should not fail, it will just pass.

Reply:
But this one is the test you were looking for: https://github.com/commercetools/commercetools-adyen-integration/pull/233/commits/7678b5946c7cc8db090fc2b264f66c22918d45be